src/model/Board.java

```diff
@@ -48,6 +48,25 @@ private void decrementPieceCount(int position) {
         }
     }
 
+    public PieceColor determineWinner() {
+        final boolean whiteHasNoPieces = this.getNumberOfWhitePieces() == 0;
+        final boolean whiteHasNoMoves = MoveGenerator.getAllPossibleMoves(this, PieceColor.WHITE)
+                .isEmpty();
+        final boolean blackHasNoPieces = this.getNumberOfBlackPieces() == 0;
+        final boolean blackHasNoMoves = MoveGenerator.getAllPossibleMoves(this, PieceColor.BLACK)
+                .isEmpty();
+
+        final boolean whiteWins = blackHasNoPieces || blackHasNoMoves;
+        final boolean blackWins = whiteHasNoPieces || whiteHasNoMoves;
+
+        if (whiteWins) {
+            return PieceColor.WHITE;
+        } else if (blackWins) {
+            return PieceColor.BLACK;
+        }
+        return null;
```

Comment:
I'd suggest throwing a `new IllegalStateException()` here with a meaningful error message. You can still put a null return after it, but if we hit this, we may as well let the runtime exception propagate up and stop execution of the program. Actually, I think we should be using these a bit more often in our code! That's probably part of the refactor I'll do on the Model code.